File: Frontend/SmartSolarMicrogrid.Web/src/types/station.ts
export interface Station {
  stationId: string;
  name: string;
  address: string;
  latitude: number;
  longitude: number;
  capacityKw: number;
  status: string;
  operatorUserId: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface StationsResponse {
  stations: Station[];
}

export interface CreateStationRequest {
  name: string;
  address: string;
  latitude: number;
  longitude: number;
  capacityKw: number;
  operatorUserId: string;
}

export type UpdateStationRequest = CreateStationRequest;

export interface NearbyStation {
  nodeId: string;
  name: string;
  latitude: number;
  longitude: number;
  distanceKm: number;
  availableSlots: number;
  status: string;
}

export interface NearbyStationsResponse {
  stations: NearbyStation[];
}

File: Android/app/src/main/java/com/smartsolar/microgrid/model/Station.kt
package com.smartsolar.microgrid.model

/**
 * Represents a solar microgrid station returned by the C# API.
 */
data class Station(
    val stationId: String,
    val name: String,
    val address: String? = null,
    val latitude: Double = 0.0,
    val longitude: Double = 0.0,
    val capacityKw: Double = 0.0,
    val status: String? = null,
    val operatorUserId: String? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null,
)

/**
 * Wrapper for GET /api/stations — the backend returns
 * `{ "stations": [ ... ] }`.
 */
data class StationsResponse(
    val stations: List<Station> = emptyList(),
)
